src/util/include/util/raii/GVariantSPtr.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Xournal++
+ *
+ * RAII wrappers for C library classes
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <utility>
+
+#include <glib-object.h>
+
+#include "util/GVariantTemplate.h"
+
+#include "CLibrariesSPtr.h"
+
+namespace xoj::util {
+
+inline namespace raii {
+namespace specialization {
+
+class GVariantHandler {
+public:
+    constexpr static auto ref = g_variant_ref;
+    constexpr static auto unref = g_variant_unref;
+    constexpr static auto ref_sink = g_variant_ref_sink;
+    constexpr static auto adopt = g_variant_take_ref;
```

Comment:
This is c++20 code, but we use c++17. Either replace constexpr with inline and make auto a const, or wrap those functions into a lambda.